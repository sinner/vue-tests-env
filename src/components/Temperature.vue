<template>
  <div>
    <span>{{ celsius }} deg. Celsius</span>
    <span>{{ fahrenheit }} deg. Fahrenheit</span>
  </div>
</template>

<script>
export default {
  props: ['temp'],

  data() {
    return {
      type: 'celsius',
      degrees: 0,
    };
  },

  computed: {
    celsius() {
      if (this.type === 'celsius') {
        return this.degrees;
      }

      return parseInt((this.degrees - 32) * 0.556, 10);
    },

    fahrenheit() {
      if (this.type === 'fahrenheit') {
        return this.degrees;
      }

      return (this.degrees * 1.8) + 32;
    },
  },

  watch: {
    temp: {
      immediate: true,
      handler(val) {
        if (!val) return;
        this.degrees = parseInt(val, 10);
        this.type = String(val).endsWith('f') ? 'fahrenheit' : 'celsius';
      },
    },
  },
};
</script>
