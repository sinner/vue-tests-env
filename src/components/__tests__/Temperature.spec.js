import { mount } from '@vue/test-utils';
import Temperature from '@/components/Temperature.vue';

describe('Temperature.vue', () => {
  it('should work properly with the celsius computed property', () => {
    const wrapper = mount(Temperature);
    expect(wrapper.vm.celsius).toBe(0);

    wrapper.setData({
      degrees: 23,
    });
    expect(wrapper.vm.celsius).toBe(23);

    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should work properly with the fahrenheit computed property', () => {
    const wrapper = mount(Temperature);
    expect(wrapper.vm.fahrenheit).toBe(32);

    wrapper.setData({
      degrees: 16,
    });
    expect(wrapper.vm.fahrenheit).toBe(60.8);

    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should set properly the temperature type property depending on the temp prop', () => {
    const wrapper = mount(Temperature, {
      propsData: {
        temp: 40,
      },
    });

    const { vm } = wrapper;
    expect(vm.degrees).toBe(40);
    expect(vm.type).toBe('celsius');

    wrapper.setProps({
      temp: '50f',
    });

    wrapper.vm.$nextTick(() => {
      expect(vm.degrees).toBe(50);
      expect(vm.type).toBe('fahrenheit');
      expect(wrapper).toMatchSnapshot();
    });
  });
});
