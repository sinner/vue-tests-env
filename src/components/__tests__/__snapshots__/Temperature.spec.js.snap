// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature.vue should set properly the temperature type property depending on the temp prop 1`] = `<div><span>10 deg. Celsius</span> <span>50 deg. Fahrenheit</span></div>`;

exports[`Temperature.vue should work properly with the celsius computed property 1`] = `<div><span>23 deg. Celsius</span> <span>73.4 deg. Fahrenheit</span></div>`;

exports[`Temperature.vue should work properly with the fahrenheit computed property 1`] = `<div><span>16 deg. Celsius</span> <span>60.8 deg. Fahrenheit</span></div>`;
